### ISSUE TITLE
Journal article not formatting properly if there is no volume field

### ISSUE THREAD

**Issue by keelancook** [dcpurton/biblatex-sbl#121]
Hello, stumbled across your work in the middle of writing my dissertation, and it's been a huge blessing! I'm in the throes of formatting my paper with our SBL proofer at my seminary. 

She noticed that one of my journal articles is not formatting like all the rest. I've discovered that the issue is the lack of a volume number in the bib entry. The particular journal does not have volume numbers. I'm just learning how to use LaTeX and biblatex, and I've no idea how to customize this fix this one entry. Any tips you have would be a big help. 

Happy to provide snippets or whatever you need for clarification.

Thanks again!

**Comment by dcpurton** [comment]
As in no parentheses around the year and no colon before the pages?

This was intentional, albeit, based on very limited data. There is no example like this in the handbook or on the blog.

The logic in my brain is that there is typically a colon between a volume and pages numbers. With journals the year comes in between, so appears in parentheses. So if there's no volume I chose to use a comma instead of a colon and drop the parentheses.

For better or worse, the style uses the same driver for the `@online` and `@article` entry types as there are similarities. This meant that I use the `@article` driver for Blogs (SBL Handbook §6.4.15). These don't have parentheses and don't have a colon.

Can you provide an example from the Chicago Manual of Style? Or ruling from SBL on how things should look like?

**Comment by keelancook** [comment]
Thanks for the quick response. What you did makes sense, and is probably what I would've chosen to do as well. As far as an example from the SBL manual, I'm lacking there as well.

Our writing center staff at our seminary say the parenthesis is required around the date, even if no volume or issue numbers are present. They claim the format is: title in italics with no following punctuation, a parenthesis around the date, followed by a colon before the page numbers. This is the same as a journal entry with the volume and issue as well, and they're telling me SBL style still calls for it without the volume. 

I'd be happy to just fix it in my files, but I'm not sure the syntax necessary to do so.

**Comment by dcpurton** [comment]
Cool. Can you please provide your bib entry so I can test?

**Comment by keelancook** [comment]
Certainly.

Here's the entry as it is in my .bib file.

```
@ARTICLE{Westerholm2020-hj,
  title   = "The Church Should Sing for Heaven's Sake: When and Why We Stopped Singing about Heaven, and How to Start Again",
  author  = "Westerholm, Matt",
  journal = "9Marks Journal",
  month   =  dec,
  year    =  2020,
  pages   = "10--20",
}
```

As I mentioned above, this particular journal has stopped using volume and issue numbers altogether. They would've made my life a lot easier had they not done so!

The current output looks like this:

![Screenshot 2021-06-30 5 38 48 AM](https://user-images.githubusercontent.com/37040841/123947014-79018880-d965-11eb-9e27-3a1c58931635.png)

The writing center gave me this correction:

![Screenshot 2021-06-30 5 40 10 AM](https://user-images.githubusercontent.com/37040841/123947151-aa7a5400-d965-11eb-8188-f90f272fc8a7.png)

Upon discussing with them, they doubled down that it should still be in parenthesis despite not having volume info. 

Thanks for reviewing this with me. Again, your template has been incredible for me. I had never used LaTeX before, and your work tipped the scales for me to try. I'm fumbling through it, but I like this so much better than what I was doing before!

**Comment by dcpurton** [comment]
I think adding this to your preamble will be sufficient:

```TeX
\renewbibmacro*{issue+date}{%
  \ifboolexpr{
    test {\iffieldundef{year}}
    and
    test {\iffieldundef{issue}}
    and
    test {\iffieldundef{date}}
    and
    test {\iffieldundef{month}}
    and
    test {\iffieldundef{day}}
  }
    {}
    {\ifentrytype{online}
       {\iffieldundef{journaltitle}
          {\newunit}
          {\setunit{\addcomma\space}}%
        \iffieldundef{issue}
          {\usebibmacro{date}}
          {\printfield{issue}%
           \setunit*{\addspace}%
           \usebibmacro{date}}
        \setunit{\addcomma\space}}
       {\setunit{\addspace}%
        \printtext[parens]{%
          \iffieldundef{issue}
            {\usebibmacro{date}}
            {\printfield{issue}%
             \setunit*{\addspace}%
             \usebibmacro{date}}}%
        \setunit{\addcolon\space}}}}
```

Can you please try for me?

**Comment by dcpurton** [comment]
*However*, without further clarification from SBL, I'm not sure if I will count this as a bug.

Consider §6.3.4 from the Handbook, which states:

> Titled book reviews should be cited as normal journal articles.
> 
> 9. Jaroslav Pelikan, “The Things That You’re Liable to Read in the Bible,” review of *The Anchor Bible Dictionary*, ed. David Noel Freedman. *New York Times Review of Books*, 20 December 1992, 3.
> 
> Pelikan, Jaroslav. “The Things That You’re Liable to Read in the Bible,” review of *The Anchor Bible Dictionary*, ed. David Noel Freedman. *New York Times Review of Books*, 20 December 1992, 3.

Although this is a book review, the handbook states that it is to be formatted as a *normal journal article*. There are no parentheses and no colon.

**Comment by keelancook** [comment]
That did the trick! 

And, I think you're probably right not to change the default behavior here. As you noted, the language in the manual seems to imply that a normal journal entry would not have the parenthesis. If so, you're initial assumptions concerning the purpose of that parenthesis being to place it between the volume/issue number the page numbers may be spot on.

This is a real edge case anyway, so I doubt it would frequently be an issue. If people run into the same need to adjust this, they can add this snippet to their preamble. 

Oh, and one last thing. I've got just a couple of other corrections that our writing center made to the style your template produced. All minor, and all things I believe I was able to adjust. Like this one, they may simply be style issues specific to my academic institution. But if you wanted  to see them (out of curiosity or for some opinion from people who deal in SBL regularly), I'm happy to pass them along. On the whole, they were very complimentary of your style!

Again, thanks!
